src/mbed_tools/build/_internal/find_files.py

```diff
@@ -66,6 +68,56 @@ def filter_files(files: Iterable[Path], filters: Iterable[Callable]) -> Iterable
     return [file for file in files if all(f(file) for f in filters)]
 
 
+class RequiresFilter:
+    """Filter out mbed libraries not needed by application.
+
+     The 'requires' config option in mbed_app.json can specify list of mbed
+    libraries (mbed_lib.json) that application requires. Apply 'requires'
+    filter to remove mbed_lib.json files not required by application.
+    """
+
+    def __init__(self, requires: Iterable[str]):
+        """Initialise the filter attributes.
+
+        Args:
+            requires: List of required mbed libraries.
+        """
+        self._requires = requires
+
+    def __call__(self, files: Iterable[Path]) -> Iterable[Path]:
+        """Return True if given path contains only allowed labels - should not be filtered out."""
```

Comment:
This docstring seems misleading. We're actually returning an `Iterable[Path]` here. Should we cast it to a `bool` if we only care about `True`/`False`?

Reply:
Fixed.